### PR TITLE
Fixes #1288: Add data structure to access get-parameters

### LINKED ISSUE #1288: Add data structure to BaseScraper to access get-parameters
- Issue type: Feature request

Add `Map` data structure that stores all get-parameters to be used.

### PR DESCRIPTION
**Pull request by vibhcool**
This PR fixes #1288

### Short description
I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by vibhcool** [comment]
@Achint08 @djmgit @hemantjadon @kavithaenair @SKrPl @singhpratyush : please review

File: src/org/loklak/harvester/BaseScraper.java [hunk 1]
```diff
@@ -35,26 +35,45 @@
...
-        this.extra = call.get("extra", "");
+        this.setExtra(call);
+        this.setParam();
         return this.getData().toJSON(false, "metadata", "posts");
     }
 
-    protected abstract Map<?, ?> getExtra(String _extra);
+    protected void setExtra(Query call) {
+        this.extra = call.getMap();
+        this.query = call.get("query", "fossasia");
```
Comment (singhpratyush) [inline src/org/loklak/harvester/BaseScraper.java]: Why default to `fossasia` here?

Comment (vibhcool) [inline src/org/loklak/harvester/BaseScraper.java]: fixed this

**Comment by SKrPl** [comment]
@vibhcool Please look into codacy issue.

**Comment by vibhcool** [comment]
@SKrPl made the suggested changes :)

File: src/org/loklak/api/search/QuoraProfileScraper.java [hunk 2]
```diff
@@ -54,10 +58,25 @@ public QuoraProfileScraper(String _query) {
...
+        this.setExtra(_extra);
+    }
+
+    protected void setParam() {
+        if("".equals(this.getExtraValue("type"))) {
+            this.typeList = Arrays.asList(this.getExtraValue("type").trim().split("\\s*,\\s*"));
+        } else {
+            this.typeList = new ArrayList<String>();
+            this.typeList.add("all");
+            this.setExtraValue("type", String.join(",", this.typeList));
```
Comment (djmgit) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: Just a doubt, here typeList contains only 1 value ("all"), so why join is being used, we can directly provide the string value

Comment (vibhcool) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: typeList is ArrayList that stores strings. if there is no get parameter, then `all` shall be set by default. 
So these lines does the task 
```
this.typeList = new ArrayList<String>();
 this.typeList.add("all");
this.setExtraValue("type", String.join(",", this.typeList));
```

Comment (vibhcool) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: This may be optimized, working on this :)

Comment (djmgit) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: Since the arraylist contains only one entry can join be avoided?

Comment (vibhcool) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: I went through it, `join()` method is needed as in map data-structure `this.extra`, all data is stored as string(the way it is fetched from get parameters) :)

**Pull request merged**